packages/core/src/Cardano/types/Transaction.ts

```diff
@@ -21,8 +57,12 @@ export interface TxBodyAlonzo {
   withdrawals?: Withdrawal[];
   certificates?: Cardano.Certificate[];
   mint?: Cardano.TokenMap;
-  scriptIntegrityHash?: Cardano.Hash16;
-  requiredExtraSignatures?: Cardano.Hash16[];
+  scriptIntegrityHash?: Cardano.Hash28ByteBase16;
+  // TODO: need to find an example of this to verify type and length
+  // This is the type we use for witness object keys too, which kinda
+  // makes sense to have as bech32, because you might want to display
+  // it to the user to show who signed the tx
```

Comment:
(Hello! I’m new to Cardano. I want to start contributing in one way or another! The future of global finances etc etc)

What about making an issue instead of a TODO comment? The issue could be linked to this PR (as a comment on this line) and anybody looking to collaborate could try to pick it up later.

I also think that bech32 sounds good for the reasons stated in this comment, so I wonder: why not just making it bech32?

Reply:
Hi @sadasant, thanks for the contribution :+1: 

We find annotating the code keeps it fresh in mind during this early development phase, with the todos reflected in our internal project management system. We're currently reserving Github issues for external submissions, but we'll move to a more traditional OS model with contributor guidelines once things stabilise. As it stands, code contributions would be considered, but we're just not actively seeking them right now.

Reply:
Hi @sadasant, thank you for a review! It is currently validated as bech32 with `ed25519_pk` prefix.

@rhyslbw some extra context for this TODO: Blockfrost doesn't seem to support `requiredExtraSignatures` field and I don't see it in cardano-graphql docs. This field is from Ogmios types and it's a `Hash16` in there.

Reply:
This Ogmios test vector has `b5ae663aaea8e500157bdf4baafd6f5ba0ce5759f7cd4101fc132f54` (56 chars):

https://raw.githubusercontent.com/CardanoSolutions/ogmios/master/server/test/vectors/ChainSync/Response/RequestNext/135.json

Reply:
Would you prefer our type to be hex as in ogmios instead of bech32?

Reply:
Sorry, no as you were. In general we should be using the bech32 version if a prefix has been defined in the CIP

Reply:
Prefix `ed25519_pk` that it currently uses is not listed in cip5 and CSL's `PublicKey.to_bech32()` does not accept a prefix argument. I think what we want is `addr_vk` prefix. I already added a dependency to `bech32` npm package so we can create the appropriate  string from `PublicKey.as_bytes()`

Reply:
(Thank you for your answers, @rhyslbw and @mkazlauskas 🙏)